helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -199,23 +211,30 @@ private WagedRebalancer(AssignmentMetadataStore assignmentMetadataStore,
         BaselineDivergenceGauge.class);
 
     _changeDetector = new ResourceChangeDetector(true);
+
+    _baselineCalculateExecutor = Executors.newSingleThreadExecutor();
+    _asyncBaselineCalculation = asyncBaselineCalculation;
   }
 
-  // Update the rebalancer preference configuration if the new preference is different from the
-  // current preference configuration.
-  public void updatePreference(
-      Map<ClusterConfig.GlobalRebalancePreferenceKey, Integer> newPreference) {
-    if (_preference.equals(NOT_CONFIGURED_PREFERENCE) || _preference.equals(newPreference)) {
-      // 1. if the preference was not configured during constructing, no need to update.
-      // 2. if the preference equals to the new preference, no need to update.
-      return;
+  // Update the rebalancer configuration if the new options are different from the current
+  // configuration.
+  public synchronized void updateRebalanceOptions(
```

Comment:
Make two methods? If client only wants to update the boolean but not update the preference, do you need the client to memorize the previous preference?

Reply:
Sounds good.